examples/llm/inference/convert_composer_to_hf.py

```diff
@@ -0,0 +1,180 @@
+        hf_tokenizer = transformers.AutoTokenizer.from_pretrained('prajjwal1/bert-tiny')
+        composer_model = HuggingFaceModel(hf_model, tokenizer=hf_tokenizer, metrics=[], use_logits=True)
+        trainer = Trainer(model=composer_model,
+                            train_dataloader=train_dataloader,
+                            save_filename='composer-hf-checkpoint.pt',
+                            max_duration='1ep',
+                            save_folder='./')
+        trainer.fit()
+        trainer.close()
+
+    Example:
+    .. testcode::
+        from composer.models import write_huggingface_pretrained_from_composer_checkpoint
+        write_huggingface_pretrained_from_composer_checkpoint('composer-hf-checkpoint.pt', './hf-save-pretrained-output')
+        loaded_model = transformers.AutoModelForSequenceClassification.from_pretrained('./hf-save-pretrained-output')
+
+    Args:
+        checkpoint_path (Union[Path, str]): Path to the composer checkpoint, can be a local path, or a remote path beginning with ``s3://``, or another backend
+            supported by :meth:`composer.utils.maybe_create_object_store_from_uri`.
+        output_folder (Union[Path, str]): Path to the folder to write the output to. Must be a local path.
+        local_checkpoint_save_location (Optional[Union[Path, str]], optional): If specified, where to save the checkpoint file to locally.
+                                                                                If the input ``checkpoint_path`` is already a local path, this will be a symlink.
+                                                                                Defaults to None, which will use a temporary file.
+    """
+    # default local path to a tempfile if path is not provided
+    if local_checkpoint_save_location is None:
+        tmp_dir = tempfile.TemporaryDirectory()
+        local_checkpoint_save_location = Path(
+            tmp_dir.name) / 'local-composer-checkpoint.pt'
+
+    # download the checkpoint file
+    get_file(str(checkpoint_path), str(local_checkpoint_save_location))
+
+    # Load the Composer checkpoint state dict
+    composer_state_dict = safe_torch_load(local_checkpoint_save_location)
+
+    # Build and save HF Config
+    print('Saving HF Model Config...')
+    hf_config = get_hf_config_from_composer_state_dict(composer_state_dict)
+    hf_config.save_pretrained(output_folder)
+    print(hf_config)
+
+    # Extract and save the HF tokenizer
+    print('#' * 30)
+    print('Saving HF Tokenizer Config...')
+    hf_tokenizer = get_hf_tokenizer_from_comopser_state_dict(
+        composer_state_dict)
+    hf_tokenizer.save_pretrained(output_folder)
+    print(hf_tokenizer)
+
+    # # Extract and save the HF model weights
+    print('#' * 30)
+    print('Saving HF Model Weights...')
+    weights_state_dict = composer_state_dict['state']['model']
+    weights_state_dict = {
+        k: v for k, v in weights_state_dict.items() if k.startswith('model.')
```

Comment:
also jira for this please

Reply:
I'm gonna remove this for now because I think it's actually a bit nicer to have this incase we have any checkpoints that have `model.` already removed